vshard/replicaset.lua

```diff
@@ -144,6 +144,49 @@ local function netbox_wait_connected(conn, timeout)
     return timeout
 end
 
+local netbox_is_conn_dead
+
+if not (util.version_is_at_least(2, 10, 0, 'beta', 2, 0) and
+        util.version_is_at_most(2, 10, 0, nil, 0, 0)) then
```

Comment:
Could you please elaborate what version range you are trying to catch here? I suppose this should get the versions where cancellation was handled correct, right? After inlining it I got `vers < 2.10.0-beta2 or ver > 2.10.0`. It also catches `2.11.0-entrypoint` which is broken and where it is not enough to check `conn.state`.

Right now we don't even know when the bug will be fixed. IMO I would be fine to make the version-agnostic checks like this:
```Lua
if conn.state == 'error' then
    return true
end
if conn.state ~= 'error_reconnect' then
    -- Neither of error states should mean it is fetching schema or doing auth,
    -- which means it is not dead.
    return false
end
if not conn._fiber then
    -- The fiber field is not present in old netbox, which correctly reports
    -- state as 'error' when its fiber got killed. It would be filtered out
    -- above then.
    return false
end
return conn._fiber:status() ~= "dead"
```
and just let the other versions fail. It isn't too critical, I think. People shouldn't kill netbox fibers in the first place. Up to you again. If you want, we can try to keep pursuing the version-specific checks. Then when we fix the bug in tnt master, we will make a follow-up patch to handle that version without the false-positive case.